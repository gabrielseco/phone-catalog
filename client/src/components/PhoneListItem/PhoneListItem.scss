@import '~styles/settings/global';
@import '~styles/tools/mixins/all';

$button-action-background: #6495ed;
$button-action-color: #ffffff;

$price-color: #a02929;

.container {
  align-items: center;
  background: #ffffff;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 1em;
  width: calc(100% - 1em);
  @include respond-to(medium) {
    width: calc(100% / 2 - 1em);
  }
  @include respond-to(large) {
    width: calc(100% / 3 - 1em);
  }
}

.container + .container {
  margin-top: 1.25em;
  @include respond-to(medium) {
    margin-top: 0;
  }
}

.container:nth-child(n + 4) {
  margin-top: 0;
  @include respond-to(medium) {
    margin-top: 2em;
  }
} 

.phoneImg {
  height: 12.5em;
  width: 12.5em;
}

.title {
  margin: 1em 0 0.5em;
}

.price {
  color: $price-color;
  font-weight: bold;
  margin-bottom: 1.5em;
}

.btnAction {
  background: $button-action-background;
  color: $button-action-color;
  display: flex;
  justify-content: center;
  padding: 0.5em;
  text-decoration: none;
  width: 10em;
}
