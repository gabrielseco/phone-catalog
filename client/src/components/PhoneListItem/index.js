export { default as PhoneListItem } from './PhoneListItem';
