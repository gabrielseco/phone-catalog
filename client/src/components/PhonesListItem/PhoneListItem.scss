@import '~styles/settings/global';
@import '~styles/tools/mixins/all';

.container {
  align-items: center;
  background: #ffffff;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 1em;
  width: calc(100% - 1em);
  @include respond-to(medium) {
    width: calc(100% / 2 - 1em);
  }
  @include respond-to(large) {
    width: calc(100% / 3 - 1em);
  }
}

.container + .container {
  margin-top: 1.25em;
  @include respond-to(medium) {
    margin-top: 0;
  }
}

.container:nth-child(n + 4) {
  margin-top: 0;
  @include respond-to(medium) {
    margin-top: 2em;
  }
} 

.phoneImg {
  height: 12.5em;
  width: 12.5em;
}

.title {
  margin: 1em 0 0.5em;
}

.price {
  color: #a02929;
  font-weight: bold;
  margin-bottom: 1.5em;
}

.btnAction {
  background: #6495ed;
  color: #ffffff;
  display: flex;
  justify-content: center;
  padding: 0.5em;
  text-decoration: none;
  width: 10em;
}
