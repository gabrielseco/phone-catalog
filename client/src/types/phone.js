// @flow
export type Phone = {
  id: number,
  name: string,
  manufacturer: string,
  price: string,
  image: string
};
