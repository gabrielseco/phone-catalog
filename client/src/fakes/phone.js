// @flow
import { type Phone } from './../types/phone';

export const phonesFake: Phone[] = [
  {
    id: 1,
    name: 'Iphone X',
    manufacturer: 'Apple Inc',
    price: '999$',
    image: 'https://img.shoppydoo.es/products/7/apple_iphone_x_64gb-300x300.jpg'
  },
  {
    id: 2,
    name: 'Samsung Galaxy S9',
    manufacturer: 'Samsung',
    price: '899$',
    image:
      'https://www.satugadget.com.my/content/images/thumbs/0008898_samsung-galaxy-s9-plus-256gb-purple-original-malaysia-set_300.jpeg'
  }
];
