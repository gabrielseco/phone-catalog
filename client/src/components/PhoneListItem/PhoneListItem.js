// @flow
import React from 'react';
import { Link, withRouter, type History } from 'react-router-dom';
import { type Phone } from './../../types/phone';
import styles from './PhoneListItem.scss';

type PhoneListItemProps = {
  phone: Phone,
  history: History
};

const PhoneListItem = (props: PhoneListItemProps) => {
  const route = `detail/${props.phone.id}`;
  const onKeyUp = evt => {
    const ENTER = 13;
    if (evt.keyCode === ENTER) {
      props.history.push(route);
    }
  };
  return (
    <div
      className={styles.container}
      onClick={() => {
        props.history.push(route);
      }}
      onKeyUp={onKeyUp}
      tabIndex={0}
      id={`phone-list-item-container-${props.phone.id}`}
    >
      <img
        className={styles.phoneImg}
        src={props.phone.image}
        alt={props.phone.name}
      />
      <h2 className={styles.title}>{props.phone.name}</h2>
      <p className={styles.price}>{props.phone.price}</p>
      <Link className={styles.btnAction} to={route}>
        See more
      </Link>
    </div>
  );
};

export { PhoneListItem as PhoneListItemTesting };

export default withRouter(PhoneListItem);
