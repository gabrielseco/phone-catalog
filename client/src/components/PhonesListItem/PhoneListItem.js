// @flow
import React from 'react';
import { Link } from 'react-router-dom';
import { type Phone } from './../../types/phone';
import styles from './PhoneListItem.scss';

type PhoneListItemProps = {
  phone: Phone
};

const PhoneListItem = (props: PhoneListItemProps) => (
  <div className={styles.container}>
    <img src={props.phone.image} alt={props.phone.name} />
    <h2 className={styles.title}>{props.phone.name}</h2>
    <div className={styles.body}>
      <p className={styles.price}>{props.phone.price}</p>
    </div>
    <Link to={`detail/${props.phone.id}`}>See more</Link>
  </div>
);

export default PhoneListItem;
