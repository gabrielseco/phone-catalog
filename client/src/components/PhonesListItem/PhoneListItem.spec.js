import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import PhoneListItem from './PhoneListItem';
import { phonesFake } from 'fakes/phone';

describe('PhoneListItem suite', () => {
  it('should render the ProductsListItem component', () => {
    const jsxComponent = (
      <Router>
        <PhoneListItem phone={phonesFake[0]} />
      </Router>
    );
    const component = shallow(jsxComponent);
    const tree = renderer.create(jsxComponent).toJSON();
    expect(component).toBeDefined();
    expect(tree).toMatchSnapshot();
  });
});
