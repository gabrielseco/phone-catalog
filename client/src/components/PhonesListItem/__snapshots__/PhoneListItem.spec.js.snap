// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneListItem suite should render the ProductsListItem component 1`] = `
<div
  className="container"
>
  <img
    alt="Iphone X"
    className="phoneImg"
    src="https://img.shoppydoo.es/products/7/apple_iphone_x_64gb-300x300.jpg"
  />
  <h2
    className="title"
  >
    Iphone X
  </h2>
  <p
    className="price"
  >
    999$
  </p>
  <a
    className="btnAction"
    href="detail/1"
    onClick={[Function]}
  >
    See more
  </a>
</div>
`;
